packages/peregrine/lib/mixins/MyAccount/useMyAccount.js

```diff
@@ -0,0 +1,27 @@
+import { useCallback } from 'react';
+import { useUserContext } from '../../context/user';
+
+const DEFAULT_TITLE = 'My Account';
+const UNAUTHED_TITLE = 'Signing Out';
+const UNAUTHED_SUBTITLE = 'Please wait...';
+
+const useMyAccount = props => {
+    const { signOut } = props;
+    const [{ currentUser }] = useUserContext();
+    const { email, firstname, lastname } = currentUser;
+    const name = `${firstname} ${lastname}`.trim() || DEFAULT_TITLE;
+    const title = email ? name : UNAUTHED_TITLE;
+    const subtitle = email ? email : UNAUTHED_SUBTITLE;
+
+    const handleSignOut = useCallback(() => {
+        signOut({ history: window.history });
+    }, [signOut]);
+
+    return {
+        handleSignOut,
+        subtitle,
+        title
+    };
+};
+
+export default useMyAccount;
```

Comment:
At least within PWA Studio, I've been making sure to export hooks as named exports, rather than `default`. Since a hook's name must start with `use` in order to work correctly, I want to encourage users to retain the name we give it, and so I put a tiny obstacle in the way of a user renaming it on import.

Reply:
Oh I noticed that. I will do that.